barretenberg/cpp/src/barretenberg/stdlib/plonk_recursion/aggregation_state/aggregation_state.hpp

```diff
@@ -92,18 +111,102 @@ template <typename Curve> void write(std::vector<uint8_t>& buf, aggregation_stat
 
     write(buf, as.P0);
     write(buf, as.P1);
-    write(buf, as.public_inputs);
-    write(buf, as.proof_witness_indices);
+    // write(buf, as.public_inputs);
+    // write(buf, as.proof_witness_indices);
     write(buf, as.has_data);
 };
 
 template <typename NCT> std::ostream& operator<<(std::ostream& os, aggregation_state<NCT> const& as)
 {
     return os << "P0: " << as.P0 << "\n"
-              << "P1: " << as.P1 << "\n"
-              << "public_inputs: " << as.public_inputs << "\n"
-              << "proof_witness_indices: " << as.proof_witness_indices << "\n"
+              << "P1: " << as.P1
+              << "\n"
+              //   << "public_inputs: " << as.public_inputs << "\n"
+              //   << "proof_witness_indices: " << as.proof_witness_indices << "\n"
               << "has_data: " << as.has_data << "\n";
 }
+template <typename Builder, typename Curve>
+aggregation_state<Curve> convert_witness_indices_to_agg_obj(Builder& builder,
+                                                            AggregationObjectIndices const& witness_indices)
+{
+    aggregation_state<Curve> agg_obj;
+
+    size_t idx = 0;
+    std::array<typename Curve::Group, 2> pairing_points;
+    for (size_t i = 0; i < 2; i++) {
+        std::array<typename Curve::BaseField, 2> base_field_vals;
+        for (size_t j = 0; j < 2; j++) {
+            std::array<typename Curve::ScalarField, 4> bigfield_limbs;
+            for (size_t k = 0; k < 4; k++) {
+                bigfield_limbs[k] = Curve::ScalarField::from_witness_index(&builder, witness_indices[idx]);
+                idx++;
+            }
+            base_field_vals[j] =
+                typename Curve::BaseField(bigfield_limbs[0], bigfield_limbs[1], bigfield_limbs[2], bigfield_limbs[3]);
+        }
+        pairing_points[i] = typename Curve::Group(base_field_vals[0], base_field_vals[1]);
+    }
+    agg_obj.P0 = pairing_points[0];
+    agg_obj.P1 = pairing_points[1];
+    return agg_obj;
+}
+
+template <typename Builder> AggregationObjectIndices init_default_agg_obj_indices(Builder& builder)
```

Comment:
creates a default agg obj and returns the 16 witness indices. will add comment to this function